### PR TITLE
ci: reduce noise generated for PRs

### PR DESCRIPTION
**Pull request by psafont**
Current CI setup generates a lot of messages for master branch which are not relevant: deprecation warnings regarding node, annotations produced for python2 code which is yet to be cleaned up, and comments with sometimes thousands of lines long after the PRs are merged.

This PR tries to remove all of them. This needs reviewing by the team doing the python3 porting to confirm this won't harm their efforts. I've adapted Bernhard's PR #5504 to include the current ignored issues to reduce friction, but it might need another pass. @liulinC @stephenchengCloud 

The changes in the LCM CI follow the ones in #5574

TODO:
- [x] Check why the Python coverage gates fail

### CODE REVIEW

**Comment by psafont** [comment]
The same tests were run, the coverage gates failing is a red herring.

File: pyproject.toml [hunk 1]
```diff
@@ -27,14 +27,74 @@ repository = "https://github.com/xapi-project/xen-api"
...
+[tool.coverage.report]
+# Here, developers can configure which lines do not need to be covered by tests:
+exclude_lines = [
+    "pragma: no cover",  # standard pragma for not covering a line or block
+    "if TYPE_CHECKING:", # imports for type checking only
+    "pass",
+    # Other specific lines that do not need to be covered, comment in which file:
+]
+# Here, developers can configure the minimum coverage required for the checks to pass:
+fail_under = 60
```
Comment (stephenchengCloud) [inline pyproject.toml:44]: Seems this is the reason why the Python coverage failed.
And there is another Python coverage configuration in .codecov.yml.

Comment (psafont) [reply pyproject.toml]: Ah, I see. This would be useful when run locally, but it's a hindrance when run on CI, I've removed it

**Comment by psafont** [comment]
I've looked at the changes from the "feature/py3" and I didn't find any contentious issues that makes this PR difficult to merge into the feature branch.
Can I get a review about it? @stephenchengCloud @liulinC

**Review by stephenchengCloud — APPROVED** [review]

**Review by lindig — APPROVED** [review]

**Review by robhoes — APPROVED** [review]

**Pull request merged**